app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm/operations.rb

```diff
@@ -15,4 +15,30 @@ def raw_create_snapshot(name, desc = nil, _memory = nil)
     create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "create")
     raise MiqException::MiqVmSnapshotError, err.to_s
   end
+
+  def raw_remove_snapshot(snapshot_id)
+    with_provider_connection(:service => 'PCloudSnapshotsApi') do |api|
+      snapshot = Snapshot.find_by(:id => snapshot_id)
+      api.pcloud_cloudinstances_snapshots_delete(cloud_instance_id, snapshot.uid_ems)
+    end
+  rescue => err
+    create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "delete")
+    raise MiqException::MiqVmSnapshotError, err.to_s
+  end
+
+  def raw_remove_all_snapshots
+    with_provider_connection(:service => 'PCloudSnapshotsApi') do |api|
+      vm = Vm.find_by(:uid_ems => ems_ref)
```

Comment:
So since this is an instance method you don't need to do a query to get `vm`, `self` is already the `vm` record.